File: SecondLab/app/src/main/java/com/example/secondlab/MainActivity.java
package com.example.secondlab;

import androidx.appcompat.app.AppCompatActivity;

import android.annotation.SuppressLint;
import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.ListView;
import android.widget.PopupMenu;
import android.widget.Toast;

import java.util.ArrayList;
import java.util.Collections;

public class MainActivity extends AppCompatActivity {

    ArrayList<Product> products = new ArrayList<>();
    ArrayList<Product> sortProducts;
    ListView productsList;
    int selectId = 0;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        Button addButton = (Button) findViewById(R.id.add_product);
        Button selectButton = (Button) findViewById(R.id.choose_workshop);


        addButton.setOnClickListener(view -> addProduct());
        selectButton.setOnClickListener(view -> selectProducts());

        Bundle bundle = getIntent().getExtras();
        if (bundle != null)
            products = (ArrayList<Product>) bundle.get("products");
        else
            setInitialData();
        updateProducts(false);
        productsList.setOnItemLongClickListener((adapterView, view, i, l) -> {
            showPopupMenu(view, i);
            return true;
        });
    }

    @SuppressLint("NonConstantResourceId")
    private void showPopupMenu(View v, int i) {
        PopupMenu popupMenu = new PopupMenu(this, v);
        popupMenu.inflate(R.menu.popupmenu);
        popupMenu.setOnMenuItemClickListener(item -> {
            switch (item.getItemId()) {
                case R.id.menu_edit:
                    Intent intent = new Intent(this,
                            EditAddActivity.class);
                    intent.putExtra("products", products);
                    intent.putExtra("product_id", i);
                    startActivity(intent);
                    return true;
                case R.id.menu_del:
                    products.remove(i);
                    updateProducts(false);
                    return true;
                default:
                    return false;
            }
        });
        popupMenu.show();
    }

    private void addProduct(){
        Intent intent = new Intent(this, EditAddActivity.class);
        intent.putExtra("products", products);
        startActivity(intent);
    }

    private void selectProducts(){
        sortProducts = new ArrayList<>();
        EditText editText = findViewById(R.id.choose_workshop_input);
        try {
            if (!editText.getText().toString().equals("")) {
                selectId = Integer.parseInt(editText.getText().toString());
                for (int i = 0; i < products.size(); i++) {
                    if (products.get(i).getWorkshopNumber() == selectId)
                        sortProducts.add(products.get(i));
                }
                if (!sortProducts.isEmpty()) {
                    Collections.sort(sortProducts, (o1, o2) ->
                            Integer.compare(o2.getCount(), o1.getCount()));
                    updateProducts(true);
                }
                else
                    Toast.makeText(getApplicationContext(),
                            "Неверные данные",
                            Toast.LENGTH_SHORT).show();
            }
        }
        catch (Exception ignored){}
    }

    private void updateProducts(Boolean select){
        productsList = findViewById(R.id.productsList);
        ProductAdapter productAdapter;
        if (!select)
            productAdapter = new ProductAdapter(this,
                    R.layout.list_item, products);
        else
            productAdapter = new ProductAdapter(this,
                    R.layout.list_item, sortProducts);
        productsList.setAdapter(productAdapter);
    }

    private void setInitialData(){
        products.add(new Product (5, 1, "Диван"));
        products.add(new Product (7, 2, "Стол"));
        products.add(new Product (1, 2, "Стул"));
        products.add(new Product (3, 3, "Шкаф"));
        products.add(new Product (12, 4, "Дверь"));
    }
}

File: MyGame/app/src/main/java/com/example/mygame/MainActivity.kt
package com.example.mygame

import android.content.Context
import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.widget.EditText
import android.widget.Toast
import androidx.appcompat.app.AlertDialog
import com.example.mygame.db.MyDbManager
import kotlinx.android.synthetic.main.activity_main.*
import java.util.*

var currentPlayer: Player? = null
val players: ArrayList<Player> = ArrayList()

class MainActivity : AppCompatActivity() {

    private val myDbManager = MyDbManager(this)

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        if(players.size == 0)
        {
            myDbManager.openDb()
            val dataList = myDbManager.readDbData()
            if(dataList.isEmpty())
            {
                myDbManager.insert("Алексей","Коновалов","10:09:38")
                dataList.add(Player("Алексей","Коновалов","10:09:38"))
                myDbManager.insert("Илья","Сиротченков","08:10:19")
                dataList.add(Player("Илья","Сиротченков","08:10:19"))
                myDbManager.insert("Дмитрий","Масленников","00:02:13")
                dataList.add(Player("Дмитрий","Масленников","00:02:13"))
            }
            for(item in dataList)
            {
                players.add(item)
            }
            myDbManager.closeDb()
        }

        btnPlay.setOnClickListener {
            if (currentPlayer != null)
            {
                btnLog.setText(R.string.btn_log_in)
                btnLog.setOnClickListener {
                    logInDialog()
                }
                val intent = Intent(this@MainActivity, PlayActivity::class.java)
                startActivity(intent)
            }
        }
        btnRecords.setOnClickListener {
            val intent = Intent(this@MainActivity, RecordsActivity::class.java)
            startActivity(intent)
        }
        btnLog.setOnClickListener {
            logInDialog()
        }
    }
    private fun logInDialog()
    {
        val builder = AlertDialog.Builder(this)
        val inflater: LayoutInflater = getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
        val view: View = inflater.inflate(R.layout.log_in_layot, null)
        builder.setView(view)
        builder.setPositiveButton("OK"
        ) { dialog, id ->
            logInPlayer(view)
            dialog.cancel()
        }
        builder.create()
        builder.show()
    }

    private fun logInPlayer(view: View)
    {
        val realName: String = (view.findViewById(R.id.editRealName) as EditText).text.toString()
        val userName: String = (view.findViewById(R.id.editUserName) as EditText).text.toString()
        if (players.indexOfFirst{it.userName == userName} == -1)
        {
            currentPlayer = Player(realName,userName,"00:00:00")
            btnLog.setText(R.string.btn_log_out)
            btnLog.setOnClickListener {
                logOutPlayer()
            }
        }
        else
        {
            Toast.makeText(applicationContext, R.string.user_name_exist, Toast.LENGTH_SHORT).show()
        }
    }
    private fun logOutPlayer()
    {
        currentPlayer = null
        btnLog.setText(R.string.btn_log_in)
        btnLog.setOnClickListener {
            logInDialog()
        }
    }
}